File: 2-ad-hoc/1140.py
while True:
    x=input().split()
    if x[0]=='*':
        break
    nao=True
    l=ord(x[0][0])
    if(l<91): dif=l+32
    elif(l>96): dif=l-32
    for _ in x:
        if ord(_[0])!=l and ord(_[0])!=dif:
            nao=False
            break
    res='N'
    res='Y' if nao else res
    print(res)

File: 2-ad-hoc/1140.cpp
#include <iostream>
#include <cstring>
// 
int main(){
    bool sim;
    char*split,res;
    int letra,dif;
    std::string *inp;
    while(true){
        sim=true;
        inp=new std::string;
        split=new char;
        std::getline(std::cin,*inp);
        if(*inp=="*"){
            break;
        }
        split=std::strtok(inp->data()," ");
        letra=int(split[0]);
        (letra<91)?dif=letra+32:(letra>96)?dif=letra-32:dif;
        for(int i=0;split!=NULL;++i){
            if(int(split[0])!=letra&&int(split[0])!=dif){
                split=NULL;
                sim=false;
                break;
            }
            split=std::strtok(NULL," ");
        }
        res='N';
        if(sim){
            res='Y';
            delete split;
            split=NULL;
        }
        std::cout<<res<<std::endl;
        delete inp;
        inp=NULL;
    }
    return 0;
}